### ISSUE TITLE
Destructuring an array's values into object member gives no-undef, no-unused-vars errors

### ISSUE THREAD

**Issue by ethanresnick** [babel/babel-eslint#115]
Here's the babel code:

```
function arrayMembersIntoObject(array, object) {
  [object.x, object.y] = array;
}
```

Running with the latest versions of babel (5.4.7), babel-eslint (3.1.9) and eslint (0.21.2), I get the following unexpected errors/warnings:

```
  1:39  warning  object is defined but never used                  no-unused-vars
  2:10  error    "x" is not defined                                no-undef
  2:20  error    "y" is not defined                                no-undef
```

When I run the compiled code through eslint directly, none of these errors occur, so I'm pretty sure this is a bug in the babel-eslint translation rather than eslint core.

**Comment by sebmck** [comment]
Seems like a default ESLint issue.

On Wednesday, 27 May 2015, Ethan <notifications@github.com> wrote:

> Here's the babel code:
>
> function arrayMembersIntoObject(array, object) {
>   [object.x, object.y] = array;
> }
>
> Running with the latest versions of babel (5.4.7), babel-eslint (3.1.9)
> and eslint (0.21.2), I get the following unexpected errors/warnings:
>
>   1:39  warning  object is defined but never used                  no-unused-vars
>   2:10  error    "x" is not defined                                no-undef
>   2:20  error    "y" is not defined                                no-undef
>
> When I run the compiled code through eslint directly, none of these errors
> occur, so I'm pretty sure this is a bug in the babel-eslint translation
> rather than eslint core.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel-eslint/issues/115>.
>


-- 
Sebastian McKenzie

**Comment by hzoo** [comment]
Can reproduce the errors with default ESLint and config

```yaml
env:
  es6: true
```

**Comment by hzoo** [comment]
Closing as it seems to be an ESLint issue. @ethanresnick feel free to open again if you find out it's only for babel-eslint.
